zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/RestoreFromBackupTool.java

```diff
@@ -618,4 +658,26 @@ private void copyProcessedRestoredFilesToDestination(FileTxnSnapLog restoreTempS
       Files.copy(processedFile.toPath(), new File(finalDestinationBase, fileName).toPath());
     }
   }
+
+  /**
+   * If the CLI command has specified a znode path to perform spot restoration,
+   * run the spot restoration tool on that path
+   * @throws IOException
+   */
+  @VisibleForTesting
+  protected void performSpotRestorationIfConfigured() throws IOException {
+    if (znodePathToRestore != null) {
+      if (zkServerConnectionStr == null) {
+        throw new IllegalArgumentException(
+            "ZK server connection info is not provided. Could not perform spot restoration.");
+      }
+      zk = new ZooKeeper(zkServerConnectionStr, CONNECTION_TIMEOUT, (event) -> {
+        System.out.println("WATCHER::" + event.toString());
+      });
```

Comment:
Could we add more logging/printing to stdout around this so the user knows whether the connection attempt was successful or not?

Reply:
We already have this kind of logs, I think it is sufficient, what do you think?
`2021-05-20 17:49:19,998 [myid:127.0.0.1:11222] - INFO  [main-SendThread(127.0.0.1:11222):ClientCnxn$SendThread@1433] - Session establishment complete on server localhost/127.0.0.1:11222, session id = 0x10017d2b5cf0000, negotiated timeout = 60000`
`2021-05-20 17:49:20,113 [myid:] - INFO  [main:ZooKeeper@1619] - Session: 0x10017d2b5cf0000 closed`

Reply:
I think it would be more helpful if we could add another line here to make it clear that it's for spot restoration.

Reply:
Added logging.